### ISSUE TITLE
Build failure: Could not resolve dependency, conflicting peer dependency

### ISSUE THREAD

**Issue by jpentland** [Ride-The-Lightning/RTL#1049]
**Describe the bug**

When building the v.12.0 tags (or current master) from source this error is shown:
npm WARN config only Use `--omit=dev` to omit dev dependencies from the install.
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: @swimlane/ngx-charts@19.2.0
npm ERR! Found: rxjs@7.4.0
npm ERR! node_modules/rxjs
npm ERR!   rxjs@"^7.1.0" from the root project
npm ERR!   peer rxjs@"^6.5.3 || ^7.4.0" from @angular/cdk@13.1.0
npm ERR!   node_modules/@angular/cdk
npm ERR!     @angular/cdk@"^13.0.2" from the root project
npm ERR!     peer @angular/cdk@"^13.0.0" from @angular/flex-layout@13.0.0-beta.36
npm ERR!     node_modules/@angular/flex-layout
npm ERR!       @angular/flex-layout@"^13.0.0-beta.36" from the root project
npm ERR!     2 more (@angular/material, @swimlane/ngx-charts)
npm ERR!   10 more (@angular/common, @angular/core, @angular/flex-layout, ...)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer rxjs@"6.x" from @swimlane/ngx-charts@19.2.0
npm ERR! node_modules/@swimlane/ngx-charts
npm ERR!   @swimlane/ngx-charts@"^19.2.0" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: rxjs@6.6.7
npm ERR! node_modules/rxjs
npm ERR!   peer rxjs@"6.x" from @swimlane/ngx-charts@19.2.0
npm ERR!   node_modules/@swimlane/ngx-charts
npm ERR!     @swimlane/ngx-charts@"^19.2.0" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See /home/user/.npm/eresolve-report.txt for a full report.
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2022-07-15T23_51_52_069Z-debug-0.log

**To Reproduce**
* npm install --only=prod

**Your environment**
* Tag v0.12.3, v0.12.2, v0.12.1 were tested. v0..11.2 does not have this issue/.
* ArchLinux, 5.18.11-arch1-1 #1 SMP PREEMPT_DYNAMIC Tue, 12 Jul 2022 15:40:51 +0000 x86_64 GNU/Linux
* npm: 8.14.0

**Comment by ShahanaFarooqui** [comment]
Please use ‘npm install —omit=dev —legacy-peer-devs’ instead ‘npm install —only=prod’. I will update the documentation with the same soon.

**Comment by jpentland** [comment]
On master branch:
```
npm install --omit=dev --legacy-peer-devs
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: @angular-eslint/builder@12.2.1
npm ERR! Found: @angular/cli@13.1.1
npm ERR! node_modules/@angular/cli
npm ERR!   dev @angular/cli@"^13.0.3" from the root project
npm ERR!   peer @angular/cli@">= 13.0.0 < 14.0.0" from @angular-eslint/schematics@13.0.1
npm ERR!   node_modules/@angular-eslint/schematics
npm ERR!     dev @angular-eslint/schematics@"13.0.1" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @angular/cli@">= 12.0.0 < 13.0.0" from @angular-eslint/builder@12.2.1
npm ERR! node_modules/@angular-eslint/builder
npm ERR!   dev @angular-eslint/builder@"^12.2.1" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: @angular/cli@12.2.17
npm ERR! node_modules/@angular/cli
npm ERR!   peer @angular/cli@">= 12.0.0 < 13.0.0" from @angular-eslint/builder@12.2.1
npm ERR!   node_modules/@angular-eslint/builder
npm ERR!     dev @angular-eslint/builder@"^12.2.1" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See /home/user/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2022-07-16T05_02_06_207Z-debug-0.log
```

**Comment by ShahanaFarooqui** [comment]
Sorry there was an iPhone auto correct typo. It is legacy-peer-deps.

**Comment by jpentland** [comment]
Yes, that works. Thanks!

**Comment by sp4ke** [comment]
The following changes since commit f84719e2c30312239e6a65fbc4662d29617dc2db:

  Removed test script from build (2022-05-01 14:44:49 -0400)

are available in the Git repository at:

  https://git.sp4ke.xyz/sp4ke/RTL-fork/ 

for you to fetch changes up to 736697e00621233bbe5e710c20aa48f661e3c524:

  Fixes #1049 npm build in dockerfiles (2022-07-21 16:35:19 +0200)

----------------------------------------------------------------
sp4ke (1):
      Fixes #1049 npm build in dockerfiles

 dockerfiles/Dockerfile         | 2 +-
 dockerfiles/Dockerfile.arm32v7 | 2 +-
 dockerfiles/Dockerfile.arm64v8 | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

**Comment by ShahanaFarooqui** [comment]
Closing after updating the documentation.
